app/GitLabRepo.php

```diff
@@ -31,6 +32,19 @@ private function __construct($url, GitLab $gitLab)
 
     public static function make($url)
     {
+        // If a user changes their name on GitLab, requests
+        // sent in a browser will redirect to the new
+        // repository while the API will just return a 404.
+        // To prevent this, we'll send an HTTP request first
+        // and see if the response is a redirect. If it is
+        // then we will use the url that the request was
+        // going to be redirected to for future requests.
+        $response = Http::withoutRedirecting()->get($url);
+
+        if ($response->status() === 301) {
+            $url = $response->headers()['Location'][0];
```

Comment:
Should we also save the updated URL locally?

Reply:
I don't think there was a reason I _didn't_ do that so yes...that would make sense. I'll include that this week.

Reply:
So the `url` ends up being updated but it doesn't really reflect that to the user since the Package model has a `url` and `repo_url` field which are used in different places.

Merging this will allow repositories that are hosted on GitLab that are also being redirected to a new url to have details like the readme updated but won't address a bigger issue of displaying that back to the user. I'm thinking we merge this and then tackle the handling of `url` vs `repo_url` separately. What do you think @mattstauffer ?

Reply:
ok sounds fine by me! :)